multipart-form-writer/progressbar.ml

```diff
@@ -0,0 +1,39 @@
+let erase_progress_bar size =
+  String.make size '\b'
```

Comment:
I think you can avoid keeping track of the size of the bar here, by using `\r` and `^K`.  Python demo:

```python
import time
import sys


def erase() -> None:
    print("\r\033[K", end="")


def put(line: str) -> None:
    erase()
    print(line, end="")
    sys.stdout.flush()


if __name__ == "__main__":
    while True:
        put("fooooooooooooo")
        time.sleep(.2)
        put("baaaar")
        time.sleep(.2)
```

Reply:
Doesn't seem to work. I replaced it with `print_string "\013\027[K"` and it doesn't erase anything.

Reply:
That's weird.  I tried it and it works for me:

```ocaml
print_string "first line\nsecond line (to be erased)";
print_string "\013\027[Knew content for second line"
```
prints
```
first line
new content for second line
```

Reply:
Dunno, I must have missed something. It works now.